# Default parameters for the demo datasets
# Documentation of the paramters can be found in the corresponding reconfigure files (see <package_name>/cfg/<package_name>/*.py)
# running rqt_reconfigure also provides tooltip text for the parameters and allows for online tuning

# General
frame_id: "odom"

# Publishers
publisher_rate: 30
pose_path_publisher_rate: 5

# Subscribers
topic_imu: "/sensor_platform/imu"
topic_baro_altimeter: "/sensor_platform/baro"
topic_radar_scan: "/sensor_platform/radar/scan"
topic_radar_trigger: "/sensor_platform/radar/trigger"

# Rosbag mode --> used by rosbag node
republish_ground_truth: True
topic_ground_truth_pose: "/ground_truth/pose"
topic_ground_truth_twist: "/ground_truth/twist"
topic_ground_truth_twist_body: "/ground_truth/twist_body"

# KF Updates
altimeter_update: True
sigma_altimeter: 5.0

radar_update: True

# Radar Measurement Model
outlier_percentil_radar: 0.05
use_w: True

## radar ego velocity estimation
# filtering
min_dist: 0.25                        # min distance of valid detection
max_dist: 100                         # Max distance of valid detection
min_db: 3                             # min SNR in [db]
elevation_thresh_deg: 60              # threshold for elevation [deg]
azimuth_thresh_deg: 60                # threshold fo azimuth [deg]
radar_velocity_correction_factor: 1.0 # Doppler velocity correction
filter_min_z: -3 # in -2 2 out
filter_max_z: 3

# zero velocity detection
thresh_zero_velocity: 0.05        # all inliers need to smaller than this value
allowed_outlier_percentage: 0.25  # outlier ratio (=percentage of detections which are allowed to be above thresh_zero_velocity)
sigma_zero_velocity_x: 0.025      # sigma v_r
sigma_zero_velocity_y: 0.025      # sigma_v_r
sigma_zero_velocity_z: 0.025      # sigma v_r

# result filtering
max_sigma_x: 0.2   # max estimated sigma to be considered an inlier (right)
max_sigma_y: 0.2  # (forward)
max_sigma_z: 0.2   # (up)
max_r_cond: 1.0e3  # max conditional number of LSQ Pseudo Inverse to ensure a stable result
use_cholesky_instead_of_bdcsvd: True # faster but less stable

# RANSAC parameters
use_ransac: True     # turn on RANSAC LSQ
outlier_prob: 0.4    # worst case outlier probability
success_prob: 0.9999 # probability of successful determination of inliers
N_ransac_points: 3   # number of measurements used for the RANSAC solution
inlier_thresh: 0.15  # inlier threshold for inlier determination

# noise offset
sigma_offset_radar_x: 0.05   # offset added to estimated sigmas
sigma_offset_radar_y: 0.025
sigma_offset_radar_z: 0.05

# ODR refinement
use_odr: True    # turn on odr refinement
min_speed_odr: 4.0 # min speed for ODR refinement
sigma_v_r: 0.125  # noise of v_r measurement used for the refinement
model_noise_offset_deg: 2.0 # min model noise
model_noise_scale_deg: 10.0 # scale model noise

# Initialization
T_init: 10
calib_gyro: true
g_n: 9.81

p_0_x: 0
p_0_y: 0
p_0_z: 0

v_0_x: 0
v_0_y: 0
v_0_z: 0

yaw_0_deg: 0

b_0_a_x: 0
b_0_a_y: 0
b_0_a_z: 0

b_0_w_x_deg: 0
b_0_w_y_deg: 0
b_0_w_z_deg: 0

b_0_alt: 0

# Initial Uncertainty
sigma_p: 0
sigma_v: 0
sigma_roll_pitch_deg: 0
sigma_yaw_deg: 0
sigma_b_a: 0.02
sigma_b_w_deg: 0.000003
sigma_b_alt: 0.1

sigma_l_b_r_x: 0.01
sigma_l_b_r_y: 0.01
sigma_l_b_r_z: 0.01
sigma_eul_b_r_roll_deg: 0.25
sigma_eul_b_r_pitch_deg: 0.25
sigma_eul_b_r_yaw_deg: 0.25


# Noise PSDs
noise_psd_a: 0.03
noise_psd_w_deg: 0.2
noise_psd_b_a: 0.00001
noise_psd_b_w_deg: 0.00001
noise_psd_b_alt: 0.000001
